FtpClient/mainwindow.cpp

```diff
@@ -51,6 +59,67 @@ void MainWindow::slotError(QAbstractSocket::SocketError err)
     }
 }
 
+void MainWindow::onChangeDir()
+{
+    std::string buffer;
+
+    buffer.resize(2);
+
+    *reinterpret_cast<short*>(&buffer[0]) = 0;
+
+    buffer += "Change dir plz";
+
+    socket->write( buffer.c_str(), buffer.size() );
+}
+
+void MainWindow::onDownloadFile()
+{
+    std::string buffer;
+
+    buffer.resize(2);
+
+    *reinterpret_cast<short*>(&buffer[0]) = 1;
+
+    buffer += "Download file plz";
+
+    socket->write( buffer.c_str(), buffer.size() );
+}
+
+void MainWindow::onUploadFile()
+{
+    QFile readStream("Your_Dir");
+    std::string directoryToSend("Your_Dir");
+
```

Comment:
We are sending directory to server to let him automatically create buffer of needed size and then file, that work's okay. QFile readStream("Your_Dir"); Here must be directory of file we want to send.